lib/realtime/tenants/janitor.ex

```diff
@@ -96,7 +84,8 @@ defmodule Realtime.Tenants.Janitor do
 
   def handle_info({:DOWN, ref, _, _, :normal}, state) do
```

Comment:
You're handling the failed case but doesn't look like the successful case. 

Seems like your state would grow unbounded with tasks over time. 

The async_nolink docs:

https://hexdocs.pm/elixir/Task.Supervisor.html#async_nolink/3

```elixir
  # The task completed successfully
  def handle_info({ref, answer}, %{ref: ref} = state) do
    # We don't care about the DOWN message now, so let's demonitor and flush it
    Process.demonitor(ref, [:flush])
    # Do something with the result and then return
    {:noreply, %{state | ref: nil}}
  end
  ```

Reply:
Oh wait... I see it now.

Reply:
I guess because you don't care about the response from the task you're handling the normal down message.